fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java

```diff
@@ -1117,7 +1120,11 @@ public void doCommit(boolean cachedObjectRequired, Context context,
                     null, null, null, obj));
 
             try { // for cleanup catch
-
+                boolean riEnabled = (m_resourceIndex != null &&
+                        m_resourceIndex.getIndexLevel() != ResourceIndex.INDEX_LEVEL_OFF);
+                @SuppressWarnings("unchecked")
+                List<Triple> before = (riEnabled && !obj.isNew()) ? m_resourceIndex.exportObject(
+                        getReader(false, null, obj.getPid())) : Collections.EMPTY_LIST;
```

Comment:
Dunno. When this finished building (love them F3 build times!) I'll import it and tinker.